go.mod

```diff
@@ -581,6 +578,33 @@ require (
 	sigs.k8s.io/structured-merge-diff/v4 v4.2.3 // indirect
 )
 
+require (
+	go.opentelemetry.io/collector/component v0.79.0
+	go.opentelemetry.io/collector/confmap v0.79.0
+	go.opentelemetry.io/collector/consumer v0.79.0
+	go.opentelemetry.io/collector/exporter v0.79.0
+	go.opentelemetry.io/collector/featuregate v1.0.0-rcv0012
+	go.opentelemetry.io/collector/processor/batchprocessor v0.79.0
+	go.opentelemetry.io/collector/processor/memorylimiterprocessor v0.79.0
+	go.opentelemetry.io/collector/receiver v0.79.0
+	go.opentelemetry.io/collector/receiver/otlpreceiver v0.79.0
+)
+
+require (
+	github.com/alecthomas/participle/v2 v2.0.0 // indirect
+	github.com/gobwas/glob v0.2.3 // indirect
+	github.com/google/s2a-go v0.1.4 // indirect
+	github.com/iancoleman/strcase v0.2.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/internal/filter v0.79.0 // indirect
+	github.com/open-telemetry/opentelemetry-collector-contrib/pkg/ottl v0.79.0 // indirect
+	github.com/shoenig/go-m1cpu v0.1.6 // indirect
+	github.com/tilinna/clock v1.1.0 // indirect
+	go.opentelemetry.io/contrib/propagators/b3 v1.17.0 // indirect
+	go.opentelemetry.io/otel/bridge/opencensus v0.39.0 // indirect
+	google.golang.org/genproto/googleapis/api v0.0.0-20230530153820-e85fd2cbaebc // indirect
+	google.golang.org/genproto/googleapis/rpc v0.0.0-20230530153820-e85fd2cbaebc // indirect
+)
```

Comment:
Yep, thanks for the feedback! I'll address the go.mod comments later on, after we have agreed on the other things. It's a bit of a pain to modify it otherwise and I don't want to waste too much time on it.

It's hard to tell why a certain thing in go.mod changed, so my process every time I rebase my branch is like this:

* I copy  paste the go.mod file from `main` to my branch
* Then I edit just the modules which I am sure I want to change, like the `opentelemetry-collector-contrib` and `opentelemetry-collector` ones
* I run `go mod tidy` so that it changes the other modules too

That's the only method I could think of which guarantees me that the change to go.mod is due to the Otel upgrade.

Reply:
(bump on this)